sfge/src/main/java/com/salesforce/graph/symbols/apex/schema/SObjectTypeChainUtil.java

```diff
@@ -0,0 +1,98 @@
+package com.salesforce.graph.symbols.apex.schema;
+
+import com.salesforce.graph.ops.ApexStandardLibraryUtil;
+import com.salesforce.graph.vertex.MethodCallExpressionVertex;
+import com.salesforce.graph.vertex.VariableExpressionVertex;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Utility to help checking chained name prefixes in Schema-based invocations. For now, this only
+ * handles expansions related to SObjectType. There could be exclusions that resolve to value types
+ * that hasn't been handled yet. TODO: Expand this to cover more chained name prefixes to improve
+ * readability in Factory classes.
+ */
+public final class SObjectTypeChainUtil {
```

Comment:
New utility to handle some chained names redundant logic. I'm restricting it to only `SObjectType` related chained names that are mostly related to `DescribeSObjectResult` factory. Wary of the side effects if I expand it too much. I've also added more test cases to various schema-related apex value tests to make sure this hasn't broken any of the existing formats.